### PR TITLE
Update to slurm-gcp 5.7.1

### PR DESCRIPTION
**Pull request by SkylerMalinowski**
### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

**Review by cboneti — COMMENTED** [review]
All tests passed. 
I left two comments, please take a look.

File: community/modules/compute/schedmd-slurm-gcp-v5-partition-dynamic/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+This module creates a dynamic compute partition that can be used as input to the
+[schedmd-slurm-gcp-v5-controller](../../scheduler/schedmd-slurm-gcp-v5-controller/README.md).
+This will configure the slurm partition to contain nodes with the corresponding feature.
+This supports externally created nodes that register as a dynamic node to also be placed
+into their corresponding partition based on node feature.
+
+> **Warning**: updating a partition and running `terraform apply` will not cause
+> the slurm controller to update its own configurations (`slurm.conf`) unless
+> `enable_reconfigure` is set to true in the partition and controller modules.
+
```
[the remark this answers is not in the document]
Comment (SkylerMalinowski) [reply community/modules/compute/schedmd-slurm-gcp-v5-partition-dynamic/README.md:12]: We can add an example after update to slurm-gcp 5.7.2.

**Review by cboneti — APPROVED** [review]

**Pull request merged**